RadioIDs.py

```diff
@@ -27,15 +32,22 @@ def _scrape(sources: List[str]) -> List[str]:
         List[str]: [description]
     """
     names: List[str] = []
+    toLookup: List[str] = []
     for source in sources:
         if source in namecache.keys():
             names.append(namecache[source])
         else:
-            # API stuff here, in progress. Even cache none so the API isn't battered
-            name = None
-            namecache[source] = name
-            names.append(name)
-    return [name for name in names if name is not None]
+            toLookup.append(source)
+
+    if toLookup:
+        response = requests.get(LOOKUP_URL, params={"radio": toLookup})
+        data = response.json()
+        log.debug(f"Data back from API: {data}")
+        for source, info in data.items():
+            namecache[str(source)] = info
+            names.append(info)
+
+    return names
 
 
 def getNames(srcList: List[dict]) -> List[str]:
```

Comment:
Looking at bot logs, I think L56 needs to be `len(names)/len(sources)`, since the list of sources could be shorter than the source list provided